### PR TITLE
feat(*): store slugs in minio & launch new pods to run builds

### PR DESCRIPTION
**Pull request by arschles**
stores code in minio and launches new pods to run [slugbuilder](https://github.com/deis/slugbuilder]. also includes experimental support for launching [dockerbuilder](https://github.com/deis/dockerbuilder) pods.

Fixes https://github.com/deis/builder/issues/7 and replaces https://github.com/smothiki/builder/pull/2/files

Related: https://github.com/deis/slugbuilder/pull/6

### CODE REVIEW

File: pkg/etcd/etcd.go [hunk 1]
```diff
@@ -65,7 +70,18 @@ type GetterSetter interface {
 // Returns:
 // 	This puts an *etcd.Client into the context.
 func CreateClient(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
-	url := p.Get("url", "http://localhost:4001").(string)
+	host := os.Getenv(hostEnvVar)
+	port, err := strconv.Atoi(os.Getenv(portEnvVar))
```
Comment (smothiki) [inline pkg/etcd/etcd.go]: This is not required as we are already doing it here 
https://github.com/deis/builder/blob/master/pkg/routes.go#L45

Comment (arschles) [inline pkg/etcd/etcd.go]: `p.Get(...)` returns an empty string when run on my cluster, so I put this in as a quick & dirty fix. I will revisit when things are stable

Comment (arschles) [inline pkg/etcd/etcd.go]: see https://github.com/deis/builder/pull/9/files#diff-534618ddc4f275f120a33a675144c889R82 below

**Comment by bacongobbler** [comment]
Looks like a lot of this PR spans multiple issues and this PR has gone too far out of scope. Would you mind breaking out these parts?

 - makefile changes
 - k8s manifest changes (imagePullPolicy, changing image name from `smothiki` to `deisci`
 - k8s secret name changes
 - go comment updates
 - anything else that is not relevant to storing slugs in minio & launching new pods to build them

This PR is small, but there's a lot of context switches and I can't tell which is which without.

**Comment by arschles** [comment]
@bacongobbler sure. I'll update here with new PRs

**Comment by arschles** [comment]
@bacongobbler new PRs created: #15, #16 (has changes for both `k8s manifest changes (imagePullPolicy, changing image name from smothiki to deisci` and `k8s secret name changes`) and #17.

**Comment by arschles** [comment]
closing as this PR has been split up as follows:

- All changes to the `Makefile` have gone into #15
- All changes to the manifest files have gone into #16
- Minor godoc changes have gone into #17
- All changes to the `builder` script, the functional changes to `etcd.go` and other godoc fixes have gone into #18

**Pull request closed without merging**